runtime/gc_base/ObjectAccessBarrier.cpp

```diff
@@ -1546,7 +1571,19 @@ MM_ObjectAccessBarrier::copyObjectFields(J9VMThread *vmThread, J9Class *objectCl
 		while (offset < limit) {
 			UDATA srcAddress = (UDATA)srcObject + srcOffset + offset;
 			UDATA destAddress = (UDATA)destObject + destOffset + offset;
-			if (sizeof(uint32_t) == referenceSize) {
+			bool copyQword = false;
+			if (isValueType
+				&& (sizeof(uint32_t) == referenceSize)
+				&& (0 == (srcAddress & 7))
+				&& ((offset + referenceSize) < limit)
+			) {
+				copyQword = true;
+			}
+			if (copyQword) {
+				Assert_MM_true(isValueType && (sizeof(uint32_t) == referenceSize));
```

Comment:
These asserts seem to be of no value - they're repeating the conditions from a few lines before.

Reply:
It is removed.